cmd/agent/subcommands/processchecks/command.go

```diff
@@ -19,11 +19,13 @@ import (
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 	processCommand.OneShotLogParams = logimpl.ForOneShot(string(command.LoggerName), "info", true)
 	checkAllowlist := []string{"process", "rtprocess", "container", "rtcontainer", "process_discovery"}
-	cmd := check.MakeCommand(
-		&processCommand.GlobalParams{
+	cmd := check.MakeCommand(func() *processCommand.GlobalParams {
+		return &processCommand.GlobalParams{
```

Comment:
Not sure I understand this change. We're passing a function instead of a variable now ?

Reply:
The thing is that when commands are registered using in this loop
https://github.com/DataDog/datadog-agent/blob/2199351fa0a74c0c5bb5ba112e8daf9a00db870a/cmd/agent/command/command.go#L96-L102

At this moment `GlobalParams` is not filled yet by the values passed to the CLI command.
So when the `processCommand.GlobalParams` struct is instantiated, it's filled by zero values.

We have to fill it dynamically after that `*command.GlobalParams` is filled (during the `command.Run` cobra call).